### PR TITLE
feat: hide 2U specific fields

### PR DESCRIPTION
**Pull request by uzairr**
Few 2U specific fields are appearing for all types of the course(s) which is creating confusion for the partners.Due to this change, 2U fields would only be displayed for 2U courses

PROD-3032

### CODE REVIEW

**Review by arbrandes — CHANGES_REQUESTED** [review]
Can we please make this a configurable setting, instead of making hard-coded references to 2U in the code?

**Review by Ali-D-Akbar — APPROVED** [review]
Do we have a snapshot where these override fields appear? 
If yes then it looks good to me.

**Review by AliAdnanSohail — APPROVED** [review]

File: src/components/EditCoursePage/EditCourseForm.jsx [hunk 1]
```diff
@@ -226,6 +227,10 @@ export class BaseEditCourseForm extends React.Component {
     this.setCollapsible(true);
   }
 
+  filterCourseTypes2U(courseType) {
```
Comment (arbrandes) [inline src/components/EditCoursePage/EditCourseForm.jsx:230]: If we remove "2U" from the object and parameter names we can avoid confusion in the future, making this a truly generic feature.

```suggestion
  filterCourseTypes(courseType) {
```

File: src/components/EditCoursePage/EditCourseForm.jsx [hunk 2]
```diff
@@ -306,6 +311,7 @@ export class BaseEditCourseForm extends React.Component {
       priceLabels,
       runTypeModes,
     } = parsedTypeOptions;
+    const courseTypes2U = courseTypeOptions.filter(this.filterCourseTypes2U);
```
Comment (arbrandes) [inline src/components/EditCoursePage/EditCourseForm.jsx:314]:
```suggestion
    const courseTypesToFilter = courseTypeOptions.filter(this.filterCourseTypes);
```

File: src/components/EditCoursePage/EditCourseForm.jsx [hunk 3]
```diff
@@ -1046,34 +1052,38 @@ export class BaseEditCourseForm extends React.Component {
...
-              optional
-            />
-            <Field
-              name="organization_logo_override_url"
-              component={ImageUpload}
-              label={(
-                <FieldLabel
-                  id="organization_logo_override.label"
-                  text="Organization Logo Override"
+            {courseTypes2U?.length > 0 && (
```
Comment (arbrandes) [inline src/components/EditCoursePage/EditCourseForm.jsx:1055]:
```suggestion
            {courseTypesToFilter?.length > 0 && (
```

File: .env [hunk 4]
```diff
@@ -1,4 +1,5 @@
 NODE_ENV='production'
+IDENTIFIER_2U='2U'
```
Comment (arbrandes) [inline .env:2]: Let's rename this as well, and remove the "2U" default setting which does not make sense for all use cases.

```suggestion
COURSE_TYPE_FILTER=''
```

File: .env.development [hunk 5]
```diff
@@ -1,4 +1,5 @@
 NODE_ENV='development'
+IDENTIFIER_2U='2U'
```
Comment (arbrandes) [inline .env.development:2]:
```suggestion
COURSE_TYPE_FILTER=''
```

File: src/components/EditCoursePage/EditCourseForm.jsx [hunk 6]
```diff
@@ -226,6 +227,10 @@ export class BaseEditCourseForm extends React.Component {
     this.setCollapsible(true);
   }
 
+  filterCourseTypes2U(courseType) {
+    return courseType.label.includes(getConfig().IDENTIFIER_2U);
```
Comment (arbrandes) [inline src/components/EditCoursePage/EditCourseForm.jsx:231]:
```suggestion
    return courseType.label.includes(getConfig().COURSE_TYPE_FILTER);
```

File: .env.test [hunk 7]
```diff
@@ -5,3 +5,4 @@ LOGO_TRADEMARK_URL=https://edx-cdn.org/v3/default/logo-trademark.svg
 LOGO_WHITE_URL=https://edx-cdn.org/v3/default/logo-white.svg
 FAVICON_URL=https://edx-cdn.org/v3/default/favicon.ico
 SITE_NAME='edX'
+IDENTIFIER_2U='2U'
```
Comment (arbrandes) [inline .env.test:8]:
```suggestion
COURSE_TYPE_FILTER=''
```

**Review by arbrandes — CHANGES_REQUESTED** [review]
Still think we can make this more generic.  Thanks for bearing with me.

**Review by ansabgillani — APPROVED** [review]

**Pull request closed without merging**